python/src/robyn/modeling/entities/clustering_results.py

```diff
@@ -1,64 +1,87 @@
 
     Attributes:
-        cluster_assignments (List[int]): Cluster assignments for each data point.
-        centroids (np.ndarray): Coordinates of cluster centroids.
-        inertia (float): Sum of squared distances of samples to their closest cluster center.
+        data (pd.DataFrame): The DataFrame containing the clustered models.
+        top_solutions (pd.DataFrame): The top solutions based on clustering.
     """
 
-    cluster_assignments: List[int]
-    centroids: np.ndarray
-    inertia: float
+    data: pd.DataFrame
+    top_solutions: pd.DataFrame
 
 
 @dataclass
-class ConfidenceIntervals:
+class ClusterCI:
     """
-    Confidence intervals for clustered data.
+    Represents the confidence intervals for the clusters.
 
     Attributes:
-        lower_bounds (pd.DataFrame): Lower bounds of confidence intervals.
-        upper_bounds (pd.DataFrame): Upper bounds of confidence intervals.
-        means (pd.DataFrame): Mean values for each cluster.
+        df_cluster_ci (pd.DataFrame): The DataFrame containing confidence intervals for the clusters.
+        boot_n (int): The number of bootstrap samples used.
+        sim_n (int): The number of simulations performed.
     """
 
-    lower_bounds: pd.DataFrame
-    upper_bounds: pd.DataFrame
-    means: pd.DataFrame
+    df_cluster_ci: pd.DataFrame
+    boot_n: int
+    sim_n: int
 
 
+@dataclass
 class ClusteringResults:
     """
-    Class to hold and manage the results of the clustering process.
+    Represents the overall results of the clustering process.
+
+    Attributes:
+        cluster_data (ClusterData): The clustered data with top solutions.
+        cluster_ci (ClusterCI): The confidence intervals for the clusters.
+        n_clusters (int): The number of clusters created.
+        errors_weights (List[float]): The weights used for error calculations.
+        clusters_means (pd.DataFrame): Mean ROI per cluster.
+        clusters_pca (pd.DataFrame): Data related to PCA clusters.
+        clusters_tsne (pd.DataFrame): Data related to t-SNE clusters.
+        correlations (pd.DataFrame): Grouped correlations per cluster.
+        plots (PlotResults): An instance of PlotResults containing all generated plots.
     """
 
-    def __init__(self, clustered_data: pd.DataFrame, confidence_intervals: ConfidenceIntervals, top_models: List[str]):
-        """
-        Initialize the ClusteringResults with the clustering outputs.
-
-        Args:
-            clustered_data (pd.DataFrame): Data with cluster assignments.
-            confidence_intervals (ConfidenceIntervals): Confidence intervals for each cluster.
-            top_models (List[str]): IDs of top performing models.
-        """
-        pass
-
-    def export_results(self, path: str) -> None:
-        """
-        Export the clustering results to files.
-
-        Args:
-            path (str): Directory path to save the results.
-        """
-        pass
+    cluster_data: ClusterData
```

Comment:
is there a distinction here between ClusterData.data and ClusteringResults.data? could we be more specific here?